pkg/cluster/k8sres.go

```diff
@@ -1273,6 +1273,12 @@ func (c *Cluster) generateCloneEnvironment(description *acidv1.CloneDescription)
 
 func (c *Cluster) generatePodDisruptionBudget() *policybeta1.PodDisruptionBudget {
 	minAvailable := intstr.FromInt(1)
+	pdbEnabled := c.OpConfig.EnablePodDisruptionBudget
+
+	// Is PodDisruptionBudget is disabled or if there is no master, set the budget to 0.
+	if (pdbEnabled != nil && !*pdbEnabled) || c.Spec.NumberOfInstances <= 0 {
```

Comment:
`c.Spec.NumberOfInstances <= 0 ` technically means there are no pods in a DB cluster; there are other situations where the master does not exists (e.g. in standby  clusters)